orchagent/dash/dashpavalidationorch.cpp

```diff
@@ -0,0 +1,364 @@
+
+    return false;
+}
+
+void DashPaValidationOrch::paValidationConfigProcess(const PaValidationEntryList& toCreate, const PaValidationEntryList& toRemove)
+{
+    SWSS_LOG_ENTER();
+
+    auto task_status = paValidationConfigApply(toCreate, toRemove);
+    if (task_status != task_success)
+    {
+        parseHandleSaiStatusFailure(task_status);
+    }
+}
+
+task_process_status DashPaValidationOrch::paValidationConfigApply(const PaValidationEntryList& toCreate, const PaValidationEntryList& toRemove)
+{
+    EntityBulker<sai_dash_pa_validation_api_t> bulker(sai_dash_pa_validation_api, gMaxBulkSize);
+    std::vector<sai_status_t> statuses;
+
+    SWSS_LOG_ENTER();
+
+    statuses.reserve(toCreate.size() + toRemove.size());
+
+    for (const auto& entry: toCreate)
+    {
+        bool new_entry = entryRefCountInc(entry);
+        if (!new_entry)
+        {
+            // mark entry as created to omit error handling
+            statuses.emplace_back(SAI_STATUS_SUCCESS);
+        } else {
+            sai_attribute_t attr;
+            uint32_t attr_count = 1;
+
+            sai_pa_validation_entry_t sai_entry;
+            sai_entry.vnet_id = entry.vnet_oid;
+            sai_entry.switch_id = gSwitchId;
+            swss::copy(sai_entry.sip, entry.address);
+
+            attr.id = SAI_PA_VALIDATION_ENTRY_ATTR_ACTION;
+            attr.value.u32 = SAI_PA_VALIDATION_ENTRY_ACTION_PERMIT;
+
+            statuses.emplace_back();
+            bulker.create_entry(&statuses.back(), &sai_entry, attr_count, &attr);
+
+            gCrmOrch->incCrmResUsedCounter(entry.address.isV4() ? CrmResourceType::CRM_DASH_IPV4_PA_VALIDATION : CrmResourceType::CRM_DASH_IPV6_PA_VALIDATION);
+        }
+    }
+
+    for (const auto& entry: toRemove)
+    {
+        bool last_ref = entryRefCountDec(entry);
+        if (!last_ref)
+        {
+            // mark entry as removed to omit error handling
+            statuses.emplace_back(SAI_STATUS_SUCCESS);
+        } else {
+            sai_pa_validation_entry_t sai_entry;
+            sai_entry.vnet_id = entry.vnet_oid;
+            sai_entry.switch_id = gSwitchId;
+            swss::copy(sai_entry.sip, entry.address);
+
+            statuses.emplace_back();
+            bulker.remove_entry(&statuses.back(), &sai_entry);
+
+            gCrmOrch->decCrmResUsedCounter(entry.address.isV4() ? CrmResourceType::CRM_DASH_IPV4_PA_VALIDATION : CrmResourceType::CRM_DASH_IPV6_PA_VALIDATION);
+        }
+    }
+
+    bulker.flush();
+
+    size_t i = 0;
+    for (; i < toCreate.size(); i++)
+    {
+        const auto& entry = toCreate[i];
+        const auto& status = statuses[i];
+
+        if (status == SAI_STATUS_SUCCESS)
+        {
+            continue;
+        }
+
+        SWSS_LOG_ERROR("Failed to create PA validation entry for VNI %u IP %s", entry.vni, entry.address.to_string().c_str());
+        task_process_status handle_status = handleSaiCreateStatus((sai_api_t) SAI_API_DASH_PA_VALIDATION, status);
```

Comment:
handle refcount here for any failures?

Reply:
The same as above, if you prefer to revert the refcount/CRM before the abort(), let me know.